packages/core/src/App/Components/Layout/Header/toggle-menu-drawer.jsx

```diff
@@ -97,7 +103,20 @@ const ToggleMenuDrawer = observer(({ platform_config }) => {
         }
 
         return () => clearTimeout(timeout.current);
-    }, [is_appstore, account_status, should_allow_authentication, is_trading_hub_category, is_next_wallet_enabled]);
+    }, [
+        is_appstore,
+        account_status,
+        should_allow_authentication,
+        is_trading_hub_category,
+        is_next_wallet_enabled,
+        is_p2p_enabled,
+    ]);
+
+    React.useEffect(() => {
+        if (is_authorize && !isSubscribed) {
+            subscribe();
+        }
+    }, [is_authorize, isSubscribed, subscribe]);
```

Comment:
I need to call subscribe in here, cashier and in p2p, because if i don't, for some reason the data from p2p_settings is not passed to these components despite being wrapped with the P2PSettingsProvider.

Reply:
it won't spam or recall the endpoint, since its safeguarded by checking if it is already subscribed to